tests/integration/test_pip_packages.py

```diff
@@ -0,0 +1,116 @@
+
+
+def test_pip_package_without_deps(test_env, tmpdir):
+    """
+    Validate data in the pip package request without dependencies.
+
+    Process:
+    Send new request to the Cachito API
+    Send request to check status of existing request
+
+    Checks:
+    * Check that the request completes successfully
+    * Check that a single pip package is identified in response
+    * Check that there are no dependencies in response
+    * Check that the source tarball includes the application source code
+    * Check that the source tarball includes an empty deps/pip directory
+    * TODO: The content manifest is successfully generated and contains correct content
+    """
+    client = Client(test_env["api_url"], test_env["api_auth_type"], test_env.get("timeout"))
+    repo = test_env["pip_packages"]["without_deps"]["repo"]
+    ref = test_env["pip_packages"]["without_deps"]["ref"]
+    pkg_managers = ["pip"]
+    initial_response = client.create_new_request(
+        payload={
+            "repo": repo,
+            "ref": ref,
+            "pkg_managers": pkg_managers,
+            # TODO: delete pip-dev-preview flag when
+            #  the pip package manager will be ready for production usage
+            "flags": ["pip-dev-preview"],
+        },
+    )
+    completed_response = client.wait_for_complete_request(initial_response)
+    assert completed_response.status == 200
+
+    assert_packages_from_response(completed_response.data, 1, pkg_managers)
+
+    assert_dependencies_from_cachito_request(completed_response.data)
+
+    # Download expected source files
+    expected_files_urls = test_env["pip_packages"]["without_deps"]["expected_files"]
+    expected_source_files = {}
+    for file_name, file_url in expected_files_urls.items():
+        expected_source_files[file_name] = get(file_url).text
+
+    # Download and extract source tarball
+    source_name = tmpdir.join(f"download_{str(completed_response.id)}")
+    file_name_tar = tmpdir.join(f"download_{str(completed_response.id)}.tar.gz")
+    resp = client.download_bundle(completed_response.id, file_name_tar)
+    assert resp.status == 200
+    assert tarfile.is_tarfile(file_name_tar)
+    with tarfile.open(file_name_tar, "r:gz") as tar:
+        tar.extractall(source_name)
+
+    # Check that the source tarball includes the application source code under the app directory.
+    assert_expected_files(path.join(source_name, "app"), expected_source_files)
+    # Check that the source tarball includes an empty deps directory.
+    assert_expected_files(path.join(source_name, "deps"))
+
+
+def assert_packages_from_response(response_data, pkg_count, pkg_managers):
+    """Check amount and type of packages in the response data."""
+    assert len(response_data["packages"]) == pkg_count
+    for idx in range(pkg_count):
+        assert response_data["packages"][idx]["type"] == pkg_managers[idx]
+
+
+def assert_dependencies_from_cachito_request(response_data, expected_deps=None):
```

Comment:
Can you describe the params as well? It would be nice to do it in all functions defined in the file.

Also, this function is the deps analogous counterpart of `assert_packages_from_response`, right? should their names match?